src/main/scala/ai/starlake/job/sink/bigquery/BigQueryJobBase.scala

```diff
@@ -308,6 +308,25 @@ trait BigQueryJobBase extends StrictLogging {
 
   protected lazy val bqNativeTable: String = BigQueryJobBase.getBqNativeTable(tableId)
 
+  /** Retry on retryable bigquery exception.
+    */
+  private def recoverBigqueryException[T](bigqueryProcess: => T, retry: Int = 0): Try[T] = {
+    Try {
```

Comment:
Did it work for you ?

Reply:
I've tested the duplicate case by forcing the call related to creation and it works but I didn't test the rate one yet. I plan to deploy a snapshot version and launch many ingestion process to see if failures happen again but it should not.

Reply:
Ok let’s just merge this and est then 
Thank you

Reply:
@hayssams @tiboun for info I'll push another PR next week to refactor some bigquery Api calls that seem to cause quota problems. We faced the issue and did a fix that is working

Reply:
Ok great 👍